actor/builtin/miner/miner_test.go

```diff
@@ -461,103 +461,344 @@ func TestMinerCommitSector(t *testing.T) {
+		minerAddr:     minerAddr,
+		currentHeight: 0,
+	}
+}
+
+func setupMinerActorLiason(t *testing.T) *minerActorLiason {
 	ctx := context.Background()
 	st, vms := th.RequireCreateStorages(ctx, t)
 
 	ancestors := th.RequireTipSetChain(t, 10)
 	origPid := th.RequireRandomPeerID(t)
 	minerAddr := th.CreateTestMiner(t, st, vms, address.TestAddress, origPid)
-	proof := th.MakeRandomPoStProofForTest()
-	done := types.EmptyIntSet()
+	return newMinerActorLiason(t, st, vms, ancestors, minerAddr)	
+}
+
+func TestMinerSubmitPoStPowerUpdates(t *testing.T) {
+	tf.UnitTest(t)
 
 	firstCommitBlockHeight := uint64(3)
 	secondProvingPeriodStart := LargestSectorSizeProvingPeriodBlocks + firstCommitBlockHeight
 	thirdProvingPeriodStart := 2*LargestSectorSizeProvingPeriodBlocks + firstCommitBlockHeight
 	fourthProvingPeriodStart := 3*LargestSectorSizeProvingPeriodBlocks + firstCommitBlockHeight
+	t.Run("power is 0 until first PoSt", func(t *testing.T) {
+		mal := setupMinerActorLiason(t)
+		
+		// commit several sectors
+		mal.requireCommit(firstCommitBlockHeight, uint64(1))
+		mal.requireCommit(firstCommitBlockHeight+1, uint64(2))
+		mal.requireCommit(firstCommitBlockHeight+1, uint64(3))
+		
+		power := mal.requirePower(firstCommitBlockHeight+2)
+		assert.Equal(t, types.NewBytesAmount(0), power)		
+	})
+	
+	t.Run("power is 1 after first PoSt", func(t *testing.T) {
+		mal := setupMinerActorLiason(t)
+		
+		// commit several sectors
+		mal.requireCommit(firstCommitBlockHeight, uint64(1))
+		mal.requireCommit(firstCommitBlockHeight+1, uint64(2))
+		mal.requireCommit(firstCommitBlockHeight+2, uint64(3))
 
-	// add a sector
-	res, err := th.CreateAndApplyTestMessage(t, st, vms, minerAddr, 0, firstCommitBlockHeight, "commitSector", ancestors, uint64(1), th.MakeCommitment(), th.MakeCommitment(), th.MakeCommitment(), th.MakeRandomBytes(types.TwoPoRepProofPartitions.ProofLen()))
-	require.NoError(t, err)
-	require.NoError(t, res.ExecutionError)
-	require.Equal(t, uint8(0), res.Receipt.ExitCode)
+		// submit PoSt and add some power.
+		done := types.EmptyIntSet()
+		mal.requirePoSt(firstCommitBlockHeight+5, done)
 
-	// add another sector
-	res, err = th.CreateAndApplyTestMessage(t, st, vms, minerAddr, 0, firstCommitBlockHeight+1, "commitSector", ancestors, uint64(2), th.MakeCommitment(), th.MakeCommitment(), th.MakeCommitment(), th.MakeRandomBytes(types.TwoPoRepProofPartitions.ProofLen()))
-	require.NoError(t, err)
-	require.NoError(t, res.ExecutionError)
-	require.Equal(t, uint8(0), res.Receipt.ExitCode)
+		power := mal.requirePower(firstCommitBlockHeight+5)
+		assert.Equal(t, types.OneKiBSectorSize, power)
+	})
+	
+	t.Run("power accumulates over multiple proving periods", func(t *testing.T) {
+		mal := setupMinerActorLiason(t)
+		
+		// Period 1 commit and prove
+		mal.requireCommit(firstCommitBlockHeight, uint64(1))
+		mal.requireCommit(firstCommitBlockHeight+1, uint64(2))
+		done := types.EmptyIntSet()
+		mal.requirePoSt(firstCommitBlockHeight+5, done)
```

Comment:
Check the power here (which will be just one sector?)

Reply:
following your advice from before (don't chain together lots of assertions instead separately set up different state) I separated this check out into "power is 1 after first post"